File: app/src/main/java/com/simpletodo/aayushf/simpletodo/TabAdapter.java
package com.simpletodo.aayushf.simpletodo;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.Log;

/**
 * Created by ayfadia on 5/15/17.
 */

public class TabAdapter extends FragmentStatePagerAdapter {
    boolean small;
    String tagtodisplay;

    @Override
    public int getItemPosition(Object object) {
        RecyclerFragment rf = (RecyclerFragment)object;
        rf.refreshFragment(tagtodisplay);
        return super.getItemPosition(object);
    }

    public String getTagtodisplay() {

        return tagtodisplay;
    }

    public void setTagtodisplay(String tagtodisplay) {
        TabAdapter.this.notifyDataSetChanged();
        if (tagtodisplay == null){
            this.tagtodisplay = " ";
        }else{
            this.tagtodisplay = tagtodisplay;
        }


    }

    public void setSmall(boolean small) {
        this.small = small;
    }

    public TabAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        Log.d("TABADAPTER", "GETITEM CALLED");
        return RecyclerFragment.newInstance(position>0, small, tagtodisplay);


    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {

        switch (position) {
            case 0:
                return "Pending";
            case 1:
                return "Done";
        }
        return null;


    }
}

File: app/src/main/java/com/simpletodo/aayushf/simpletodo/TabAdapter.kt
package com.simpletodo.aayushf.simpletodo

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.util.Log

/**
 * Created by ayfadia on 5/15/17.
 */

class TabAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    internal var small: Boolean = false
    internal  var tagtodisplay = ""

    override fun getItemPosition(`object`: Any?): Int {
        val rf = `object` as RecyclerFragment?
        rf!!.refreshFragment(tagtodisplay)
        return super.getItemPosition(`object`)
    }

    fun getTagtodisplay(): String {

        return tagtodisplay
    }

    fun setTagtodisplay(tagtodisplay: String?) {
        this@TabAdapter.notifyDataSetChanged()
        if (tagtodisplay == null) {
            this.tagtodisplay = " "
        } else {
            this.tagtodisplay = tagtodisplay
        }


    }

    fun setSmall(small: Boolean) {
        this.small = small
    }

    override fun getItem(position: Int): Fragment {
        Log.d("TABADAPTER", "GETITEM CALLED")
        return RecyclerFragment.newInstance(position > 0, small, tagtodisplay)


    }

    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence? {

        return when (position) {
            0 -> "Pending"
            1 -> "Done"
            else -> {
                return null

            }
        }


    }
}
